kickoff/model.py

```diff
@@ -340,36 +359,67 @@ def getReleases(ready=None, complete=None, shipped=None, productFilter=None,
     return releases
 
 
-def getOrderByList(obj, orderByDict={}):
-    lst = []
-
-    for k, v in orderByDict.iteritems():
-        column = getattr(obj, k)
-        direction = getattr(column, v)
-        lst.append(direction())
-
-    return lst
+release_union = polymorphic_union({
+    'firefox': FirefoxRelease.__table__,
+    'fennec': FennecRelease.__table__,
+    'thunderbird': ThunderbirdRelease.__table__
+}, 'product', 'release')
 
 
-class ProductReleasesView(Release, db.Model):
-    __tablename__ = 'product_releases'
-
-    product = db.Column(db.String(100))
-    partials = db.Column(db.String(100))
-    promptWaitTime = db.Column(db.Integer(), nullable=True)
-    commRevision = db.Column(db.String(100))
-    commRelbranch = db.Column(db.String(50))
+class ProductReleasesView(object):
+    view = release_union
 
     @classmethod
-    def OR(cls, searchList):
-        return or_(*searchList)
+    def getOrderByList(cls, orderByDict):
+        lst = []
+        for k, v in orderByDict.iteritems():
+            column = getattr(cls.view.c, k)
+            direction = getattr(column, v)
+            lst.append(direction())
+
+        return lst
 
     @classmethod
-    def getSearchList(cls, searchDict={}):
+    def getSearchList(cls, searchDict):
         lst = []
         for k, v in searchDict.iteritems():
-            column = getattr(cls, k)
+            column = getattr(cls.view.c, k)
             contains = getattr(column, 'contains')
             lst.append(contains(v))
 
         return lst
+
+    @classmethod
+    def getTotal(cls, complete, ready, searchFilter):
+        andFilter = {}
+        if ready is not None:
+            andFilter['ready'] = ready
+        if complete is not None:
+            andFilter['complete'] = complete
+
+        qry = db.session.query(cls.view)
+
+        if andFilter:
+            qry = qry.filter(and_(*cls.getSearchList(andFilter)))
+        if searchFilter:
+            qry = qry.filter(or_(*cls.getSearchList(searchFilter)))
+
+        return qry.count()
+
+    @classmethod
+    def query(cls):
+        return db.session.query(cls.view)
+
+    @classmethod
+    def releaseToDict(cls, release):
+        d = {}
+        for i, c in enumerate(cls.view.c):
+            d[c.name] = release[i]
+
+        if d['submittedAt']:
+            d['submittedAt'] = pytz.utc.localize(d['submittedAt']).isoformat()
+
+        if d['shippedAt']:
+            d['shippedAt'] = pytz.utc.localize(d['shippedAt']).isoformat()
+
+        return d
```

Comment:
IT would be great to add some tests for these methods. They involve some magic and it may be very easy to break them.